client_app/src/components/SearchBar/SearchBar.jsx

```diff
@@ -0,0 +1,46 @@
+import {React, useRef} from "react";
+import {IconButton, InputBase} from "@material-ui/core";
+import SearchIcon from "@material-ui/icons/Search";
+import useStyles from "./styles";
+import {useHistory} from "react-router-dom";
+import {ROUTES} from "../../utils/enums";
+import {useTranslation} from "react-i18next";
+
+export default function SearchBar(){
+  const classes = useStyles();
+  const history = useHistory();
+  const input = useRef();
+  const {t} = useTranslation();
+
+  function handleSubmitSearch(){
+    history.push(ROUTES.SEARCH + "?query=" + input.current.value);
```

Comment:
devrait etre un state non ? 

`const [currentInput, setCurrentInput] = useSate("");`

Reply:
Ouin j'y ai pensé, j'me disait que c'était pt plus optimisé de pas modifier un state à chaque input mais c'est vrai que ce serait plus react comme façon de faire

Reply:
React dit de ne pas utiliser des `refs` si on peut utiliser un state dans leur doc mais je ne me rappel pu ou

Reply:
https://legas314problog.wordpress.com/2020/11/07/useref-vs-usestate/
Je vais garder useRef pour une question d'optimisation et aussi s'assurer que la valeur passée soit toujours la bonne.
> 1. If you want to track your variable’s value and make a re-render when this happens, go for UseState. But be aware this can have a performance downside on your app, calling many re-renders changing multiples useStates can create bottle necks.
>2. There’s another important point to consider when comparing UseState and UseRef and it’s when they’re executed: UseState is an Asynchronous hook, meaning you can expect to modify it’s value in one line and be updated in the next one; UseRef, in the other hand, it’s a Synchronous hook, which means that the running code actually “waits” for the Ref’s value to be modified.